### ISSUE TITLE
Ubuntu 22.04/24.04 upgrade

### ISSUE THREAD

**Issue by TT-billteng** [tenstorrent/tt-metal#10469]
Our recent upgrade to C++20 has required switching to clang-17 + libc++ in order for our code to compile correctly. This isn't great as it's usually a good idea to be compatible with  the compiler toolchain native to an OS distro. The toolsets available on Ubuntu 20.04 are too old as we need at least g++12 and beyond, thus forcing the switch to Ubuntu 22.04. Steps in order to make this happen:

- [x] 22.04 docker for metal compilation
- [ ] #12488
- [ ] #12490
- [ ] #12491
- [x] Upgrade ansible scripts to provision a Ubuntu 22.04 VM runner with KMD/firmware/Metal dependencies
- [ ] #12492

**Comment by marty1885** [comment]
Is there any reason to not use 24.04? I believe most end uses would be on the latest LTS and having CI that makes sure that version runs smoothly is critical for DX.

**Comment by TT-billteng** [comment]
could do it, but currently can't create native 24.04 instances on our internal cloud

**Comment by TT-billteng** [comment]
Also, tt-BUDA has 20.04 and 22.04 docker images so I'd like to stay in sync for now

**Comment by marty1885** [comment]
Thanks for the explanation, totally understandable. I was hoping TT can move to 24.04 so I wont need spend as much time debugging build issues on the latest GCC. I'll do my best then.

**Comment by TT-billteng** [comment]
> Thanks for the explanation, totally understandable. I was hoping TT can move to 24.04 so I wont need spend as much time debugging build issues on the latest GCC. And can more focus on debugging my code.. I'll do my best then.

I'll look into adding Ubuntu 24.04 into CI, at least compiling with 24.04's GCC. The main problem we have is that we don't have dedicated builder machines in our internal cloud yet, and we don't want to pay for extra GH runner capacity :|.

**Comment by marty1885** [comment]
> and we don't want to pay for extra GH runner capacity

That is new to me. I might missed something but AFAIK GH hosted runners on public repositories are completely free. And projects I maintain uses them extensively without paying a dime. `tt-metal` should be eligible. See [About billing for GitHub Actions](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions)

Quote:
> GitHub Actions usage is free for standard GitHub-hosted runners in public repositories

**Comment by TT-billteng** [comment]
> > and we don't want to pay for extra GH runner capacity
> 
> That is new to me. I might missed something but AFAIK GH hosted runners on public repositories are completely free. And projects I maintain uses them extensively without paying a dime. `tt-metal` should be eligible. See [About billing for GitHub Actions](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions)
> 
> Quote:
> 
> > GitHub Actions usage is free for standard GitHub-hosted runners in public repositories

They are but we get a maximum of 60 instances. We have a lot of people running CI jobs 😓

**Comment by TT-billteng** [comment]
One of things that could help a lot is device arch independent builds, hopefully will get around to it soon.

**Comment by marty1885** [comment]
Ah I see... hope we can get 24.04 some time soon then. I'll keep posting patches in the mean time.

**Comment by TT-billteng** [comment]
> Ah I see... hope we can get 24.04 some time soon then. I'll keep posting patches in the mean time.

We've found a nice method to boost our building capacity + more local builders so this should be possible once our docker workflow infra is in place

**Comment by TT-billteng** [comment]
closing in favor of https://github.com/tenstorrent/tt-metal/milestone/8
